### DISCUSSION TITLE
Why resetting a form using form method without specifying field name doesn't work.

### DISCUSSION THREAD

**Discussion by jeepers3327** [react-hook-form/react-hook-form#8576] [Q&A] [answered]
I'm curious why does `reset` method doesn't  work with using it without passing any parameters on it?

```
  const {
    register,
    handleSubmit,
    formState: { errors },

    reset,
    watch,
  } = useForm<AddCommentSchema>({
    resolver: zodResolver(schema),
  });

 <VStack
      my="4"
      as="form"
      alignItems="flex-start"
      onSubmit={handleSubmit((data) => {
        console.log(data);
        //reset({ body: '' });
        //reset();
      })}
    >
  <Textarea maxLength={600} rows={2} {...register('body')} />
</VStack>
```

when I use `reset()` it doesn't reset the value, the previous value still remain but when I call reset with specified field name it clears out the value`reset({ body: '' });`

**Comment by bluebill1049** [comment, marked as answer]
options:

- you need to provide a `defautValues` react hook form can't guess the defaultValues for your form
- supply form values for reset function

**Comment by Upinwv** [comment]
Canary Mail
You've received a secure email
***@***.*** has sent you a secure email via Canary Mail.

Read Secure Email (https://secure.canarymail.io/read?obj_id=714be78e-6f15-4654-945c-d74cb5eee780&obj_key=MlJsL1huTmNMMC82TUtlWnNNTHpTdz09&thr_id=714be78e-6f15-4654-945c-d74cb5eee780)

If you expect to correspond often with ***@***.***, we recommend you download Canary Mail for free.

Download Canary (https://canarymail.io)

Privacy (https://canarymail.io/privacy.html) | Terms (https://canarymail.io/terms.html) | Docs (https://help.canarymail.io/) | Support (https://canarymail.zendesk.com/hc/en-us/requests/new)

Copyright © 2021 Canary Mail, All rights reserved.
